### PR TITLE
Add support for kubernetes version 1.30.0

### PR DESCRIPTION
**Pull request by MichaelKatsoulis**
## Proposed commit message
Add support to 1.30 Kubernetes version:

- Update documentation
- Change K8s cluster version to 1.30
- Update controller manager, scheduler and proxy metrics files.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Relates to [ #3068](https://github.com/elastic/observability-dev/issues/3068)

### CODE REVIEW

**Comment by MichaelKatsoulis** [comment]
Note:

**Comment by MichaelKatsoulis** [comment]
The PR is now ready for review!
@fearful-symmetry or @faec  could you also review it as part of data plane team?

File: deploy/kubernetes/Jenkinsfile.yml [hunk 1]
```diff
@@ -18,5 +18,5 @@ stages:
             make check-no-changes;
         stage: checks
     k8sTest:
-        k8sTest: "v1.29.0,v1.28.0,v1.27.3,v1.26.6"
+        k8sTest: "v1.30.0,v1.29.0,v1.28.0,v1.27.3"
```
Comment (gizas) [inline deploy/kubernetes/Jenkinsfile.yml:21]: Can we update the list based on the https://kubernetes.io/releases/?

Comment (MichaelKatsoulis) [reply deploy/kubernetes/Jenkinsfile.yml:21]: As kind is used to spin up the kubernetes clusters, we can only use the available kindest/node as found in https://hub.docker.com/r/kindest/node/tags?page=&page_size=&ordering=&name=.

The latest for 1.27 available is v1.27.13
For 1.28 the latest available  is v1.28.9
For 1.29 the latest available is v1.29.4.

Do you suggest to use those or keep the ones we already have?

Comment (gizas) [reply deploy/kubernetes/Jenkinsfile.yml:21]: The latter the better then. So I would say to update them

**Review by tetianakravchenko — COMMENTED** [review]
in those files is defined `K8S_VERSION`:
https://github.com/elastic/beats/blob/main/.buildkite/deploy/kubernetes/deploy-k8s-pipeline.yml
https://github.com/elastic/beats/blob/main/.buildkite/metricbeat/pipeline.yml
https://github.com/elastic/beats/blob/main/.buildkite/filebeat/filebeat-pipeline.yml
https://github.com/elastic/beats/blob/main/Jenkinsfile
should it be also updated?

**Comment by tetianakravchenko** [comment]
Please update documentation https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/_meta/docs.asciidoc#compatibility

**Review by cmacknz — APPROVED** [review]

**Review by gizas — APPROVED** [review]

**Comment by MichaelKatsoulis** [comment]
> in those files is defined `K8S_VERSION`: https://github.com/elastic/beats/blob/main/.buildkite/deploy/kubernetes/deploy-k8s-pipeline.yml https://github.com/elastic/beats/blob/main/.buildkite/metricbeat/pipeline.yml https://github.com/elastic/beats/blob/main/.buildkite/filebeat/filebeat-pipeline.yml https://github.com/elastic/beats/blob/main/Jenkinsfile should it be also updated?

These tests are new.  @oakrizan should we update those files, testing with the new kubernetes version as part of this PR ?

**Comment by MichaelKatsoulis** [comment]
This PR scope is to update and test the support of kubernetes version 1.30.0 for kubernetes metricbeat module . The tests and files of https://github.com/elastic/beats/pull/39210#pullrequestreview-2141720019 are not maintained by elastic/obs-ds-hosted-services team. So I will leave them out of this PR.

**Pull request merged**